gobblin-temporal/src/main/java/org/apache/gobblin/temporal/ddm/workflow/impl/ExecuteGobblinWorkflowImpl.java

```diff
@@ -178,32 +176,31 @@ private void cleanupWorkDirs(WUProcessingSpec workSpec, EventSubmitterContext ev
     try {
       CleanupResult cleanupResult = deleteWorkDirsActivityStub.delete(workSpec, eventSubmitterContext,
           calculateWorkDirsToDelete(jobState.getJobId(), directoriesToClean));
-      if (directoriesToClean.size() != cleanupResult.getAttemptedCleanedDirectories().size()) {
-        log.warn("Expected to clean up {} directories, but only cleaned up {}", directoriesToClean.size(),
-            cleanupResult.getAttemptedCleanedDirectories().size());
-        for (String dir : directoriesToClean) {
-          if (cleanupResult.getAttemptedCleanedDirectories().get(dir)) {
-            log.error("Directory {} was not cleaned up, please clean up manually", dir);
-          }
+      for (String dir : directoriesToClean) {
+        if (!cleanupResult.getDeletionSuccessesByDirPath().get(dir)) {
+          log.error("Directory {} was not cleaned up, please clean up manually", dir);
```

Comment:
shouldn't this be an exception?

Reply:
I don't think we should be throwing an exception here if the cleanup skips a non-job scoped dir, in case jobs with directories that do not use the dataset handler service are run in GaaS. Otherwise there will be a flood of jobs that should be passing that are now failing. There's a consideration that we should have a configuration to override the job scoped requirement there is currently there.